scst-store/logs.md

```diff
@@ -0,0 +1,99 @@
+
+##### Common to all logs
+The following key/value pairs are common for all logs.
+
+| Key | Type | Log Level | Description |
+| --- | ---- | ----------| ------------|
+| msg | string | default | A short description of the event being logged. In the case of error log line, these will also include the error message. |
+| endpoint | string | default | The API endpoint the Metadata Store is attempting to handle the request for. This also includes any query and path parameters passed in. |
+| hostname | string | default | The kubernetes hostname of the pod handling the request. This helps identify the specific instance of the Store when multiple instances of it is deployed on a cluster. |
+| function | string | debug | The function name that handles the request. |
+| method | string | default |The HTTP verb used to access the endpoint. For example, "GET" or "POST" |
+| code | integer | default | The HTTP response code |
+| response | string | default | The HTTP response in human readable format. For example, "OK", "Bad Request", or "Internal Server Error" |
+
+##### Logging Query and Path Parameter values
+
+For those endpoints that use query or path parameters, these parameters are logged on the `Request parameters` log line as key/value pairs. Afterwards, they are appended to all other log lines of the same request as key/value pairs.
+
+The key name are the name of the query or path parameter, while the value are set to the value of those parameters, in string format.
+
+For example, the following log line
+```text
+I1122 20:30:21.869791       1 images.go:34] MetadataStore "msg"="Request parameters" "endpoint"="/api/images?digest=sha256%3A20521f76ff3d27f436e03dc666cc97a511bbe71e8e8495f851d0f4bf57b0bab6" "hostname"="metadata-store-app-564f8995c8-r8d6n" "method"="GET" "digest"="sha256:20521f76ff3d27f436e03dc666cc97a511bbe71e8e8495f851d0f4bf57b0bab6" "id"=0
+```
+contains the `digest` and `id` key, which represents the respective `digest` and `id` query parameters, as well as their values. 
+
+These key/value pairs shows up in all subsequent log lines of the same call. For example:
+```text
+I1122 20:30:21.878749       1 images.go:56] MetadataStore "msg"="Request response" "digest"="sha256:20521f76ff3d27f436e03dc666cc97a511bbe71e8e8495f851d0f4bf57b0bab6" "endpoint"="/api/images?digest=sha256%3A20521f76ff3d27f436e03dc666cc97a511bbe71e8e8495f851d0f4bf57b0bab6" "hostname"="metadata-store-app-564f8995c8-r8d6n" "id"=0 "method"="GET" "code"=200 "response"="OK"
+```
+
+This is done to ensure two things:
+1. The application interprets the values of the query/path parameters correctly
+2. Help figure out which log lines are associated with a particular API request. Since there can be a number of simultaneous endpoint calls, this is a first attempt at trying to group logs by specific calls. An improved way to identify and group log by specific request is in the work for a future release.
```

Comment:
```suggestion
2. Help figure out which log lines are associated with a particular API request. Since there can be a number of simultaneous endpoint calls, this is a first attempt at trying to group logs by specific calls.
```

Reply:
Probably don't need to mention future plans in this doc unless it's very helpful.